src/dataServices/dynamoDbDataService.ts

```diff
@@ -97,13 +99,16 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource } = request;
+        return this.createResourceWithId(resourceType, resource, uuidv4());
+    }
 
+    private async createResourceWithId(resourceType: string, resource: any, resourceId: string) {
         const vid = 1;
         let item = resource;
         item.resourceType = resourceType;
         item.meta = generateMeta(vid.toString());
 
-        const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, uuidv4(), vid);
+        const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, resourceId, vid);
         try {
             await this.dynamoDb.putItem(params).promise();
         } catch (e) {

```

Comment:
on line 117 Can you re-word the error message to be more accurate; perhaps: 

"Cannot create resource since the id matched an existing id"